src/main/java/de/tum/in/www1/artemis/service/ExamService.java

```diff
@@ -297,4 +302,24 @@ private Exercise selectRandomExercise(SecureRandom random, ExerciseGroup exercis
         int randomIndex = random.nextInt(exercises.size());
         return exercises.get(randomIndex);
     }
+
+    /**
+     * Starts all the exercises of all the student exams of an exam
+     *
+     * @param examId exam to which the student exams belong
+     * @return list of generated participations
+     */
+    @Transactional
+    public List<Participation> startExercises(Long examId) {
+        List<StudentExam> studentExams = studentExamRepository.findByExamId(examId);
+        List<Participation> generatedParticipations = new ArrayList<>();
+
+        for (StudentExam studentExam : studentExams) {
+            User student = studentExam.getUser();
+            studentExam.getExercises().stream().filter(exercise -> exercise.getStudentParticipations().isEmpty())
```

Comment:
exercise.getStudentParticipations().isEmpty() is problematic here, because it might include participations of other users than the user specified in the studentExam

Reply:
I'll add comments and we have to improve this later on, but as others need this functionality for testing, we'll merge this now